diff --git a/rte/CMakeLists.txt b/rte/CMakeLists.txt
index f89a917..4c69e1b 100644
--- a/rte/CMakeLists.txt
+++ b/rte/CMakeLists.txt
@@ -82,10 +82,9 @@ if(BUILD_STATIC_LIBS)
   find_package(TIFF REQUIRED)
   find_package(LZMA REQUIRED)
   find_package(zstd REQUIRED)
-  find_package(JBIG REQUIRED)
   find_package(Deflate REQUIRED)
   find_package(WebP REQUIRED)
-  target_link_libraries(SDL2::Image INTERFACE ${JPEG_LIBRARIES} ${PNG_LIBRARIES} ${TIFF_LIBRARIES} ${LZMA_LIBRARIES} ${zstd_LIBRARIES} ${JBIG_LIBRARIES} ${Deflate_LIBRARIES} ${WebP_LIBRARIES} -lLerc -lwebpmux -lwebpdemux -lsharpyuv)
+  target_link_libraries(SDL2::Image INTERFACE ${JPEG_LIBRARIES} ${PNG_LIBRARIES} ${TIFF_LIBRARIES} ${LZMA_LIBRARIES} ${zstd_LIBRARIES} ${Deflate_LIBRARIES} ${WebP_LIBRARIES} -lLerc -lwebpmux -lwebpdemux -lsharpyuv)
 
   # SDL2::Mixer libraries
 
